### PR TITLE
Add run time information and update fv3 to use unified missing value

### LINKED ISSUE #57: File to indicate output file is fully written
So that the workflow does not trigger post jobs or use files that are only partially written, a log file to indicate that a file is fully written is requested.  See https://github.com/NOAA-EMC/global-workflow/issues/935 for more explanation as to why.

### LINKED ISSUE #64: Update CICE to latest Consortium/main
Updates made at Consortium include

- Fixes for advanced snow physics. Includes both Icepack updates (https://github.com/CICE-Consortium/Icepack/pull/449) and corresponding CICE update (https://github.com/CICE-Consortium/CICE/pull/852). These changes include a bugfix in Icepack for the calculation of smicen and smliqn. 

- Deprecation of ``zsalinity`` (https://github.com/CICE-Consortium/CICE/pull/851)

### LINKED ISSUE #677: Adding run time information for fv3 grid component
## Description
This PR is to add run time information at each time step in fv3 cap for the NUOPC phases and upp updates.

## Solution
1. Add timing information in the fv3 cap.
2. For UPP: 
(1) make upp revision up-to-date
(2) add printout information for inline post 2d decomposition
(3) tweak reading microphysics number concentration
(4) read 3D radar reflectivity when model uses Thmopson MP

## Alternatives
If applicable, add a description of any alternative solutions or features you've considered.

## Testing:

1. Have you tested the code changes? On what platforms?
on hera
2. Have you run regression test in ufs-weather-model code changes?
- Will the baseline results change?
- If yes, please give brief explanation of your code updates on the regression test.

## Dependent PRs:

Directly reference any issues or PRs in this or other repositories that this is related to, and describe how they are related. Example:
- associated https://github.com/ufs-community/ufs-weather-model/pull/1823

### LINKED ISSUE #1782: Timing information for the coupled runs
## Description
To analyze the temporary load balance issue in the coupled model runs, it is required that the components models provide timing information for the initial, run and finalize phases. We need to add timing information n a consistent way in the model component nuopc caps. 

## Solution
Provide timing information in the initial, run and finalize phases in all the model components.

### PR DESCRIPTION
**Pull request by junwang-noaa**
## Description
This PR will add run time information  in FV3ATM, CICE6 and MOM6 component and update fv3atm to use unified missing value

### Input data additions/changes
- [x] No changes are expected to input data.
- [ ] Changes are expected to input data:
  - [ ] New input data.
  - [ ] Updated input data.

### Anticipated changes to regression tests:
- [x] No changes are expected to any regression test.
- [ ] Changes are expected to the following tests:  All the RT tests with atmf history output files will change results.

## Subcomponents involved:
- [ ] AQM
- [ ] CDEPS
- [x] CICE
- [ ] CMEPS
- [ ] CMakeModules
- [x] FV3
- [ ] GOCART
- [ ] HYCOM
- [x] MOM6
- [ ] NOAHMP
- [ ] WW3
- [ ] stochastic_physics
- [ ] none

### Library Updates/Changes

- [x] Not Needed
- [ ] Create separate issue in [JCSDA/spack-stack](https://github.com/JCSDA/spack-stack) asking for update to library. Include library name, library version.
- [ ] Add issue link from JCSDA/spack-stack following this item

## Commit Queue Checklist:

- [x] Link PR's from all sub-components involved in section below
- [x] Confirm reviews completed in ALL sub-component PR's
- [x] Add all appropriate labels to this PR.
- [x] Run full RT suite on either Hera/Cheyenne AND attach log to a PR comment.
- [x] Add list of any failed regression tests to "Anticipated changes to regression tests" section.

## Linked PR's and Issues:

Dependent PRs:
- Depends on https://github.com/NOAA-EMC/fv3atm/pull/678
- Depends on https://github.com/NOAA-EMC/CICE/pull/65
- Depends on https://github.com/NOAA-EMC/MOM6/pull/114
- Closes https://github.com/NOAA-EMC/fv3atm/issues/677
- Closes https://github.com/ufs-community/ufs-weather-model/issues/1782
- Closes https://github.com/NOAA-EMC/CICE/issues/64
- Closes https://github.com/NOAA-EMC/CICE/issues/57

## Testing Day Checklist:

- [x] This PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR.
- [ ] Move new/updated input data on RDHPCS Hera and propagate input data changes to all supported systems.

### Testing Log (for CM's):
- RDHPCS
  - [x] Hera
  - [x] Orion
  - [x] Jet
  - [x] Gaea
  - [x] Cheyenne
- WCOSS2
  - [x] Dogwood/Cactus
  - [ ] Acorn
- CI
  - [ ] Completed
- opnReqTest
  - [ ] N/A
  - [x] Log attached to comment

### CODE REVIEW

File: tests/parm/nems.configure.cpld_noaero_outwav.IN
Comment (DeniseWorthen) [inline tests/parm/nems.configure.cpld_noaero_outwav.IN:124]: Is this the only test we want to add this attribute? This is the waves in the slow loop test. We're not adding waves for most of the scalability testing.

Comment (junwang-noaa) [reply tests/parm/nems.configure.cpld_noaero_outwav.IN:124]: I am trying to add the runtime log to the GFSv17 test, cpld_control_gfsv17 test, which is using this nems.configure file.

**Comment by junwang-noaa** [comment]
The full RT passed on hera, the log file is at:

/scratch1/NCEPDEV/nems/Jun.Wang/nems/vlab/20230705/ufs-weather-model/tests/logs/RegressionTests_gaea.log_20230828

**Comment by BrianCurtis-NOAA** [comment]
> The full RT passed on hera, the log file is at:
> 
> /scratch1/NCEPDEV/nems/Jun.Wang/nems/vlab/20230705/ufs-weather-model/tests/logs/RegressionTests_gaea.log_20230828

Did you mean to link to the Gaea log?

**Comment by junwang-noaa** [comment]
Good catch! I copied the wrong log file to this temporary file, now I lost my hera log file. I will run full RT again..

**Comment by junwang-noaa** [comment]
The hera RT log file is at: 
/scratch1/NCEPDEV/nems/Jun.Wang/nems/vlab/20230705/ufs-weather-model/tests/logs/RegressionTests_hera.log_20230830

**Comment by FernandoAndrade-NOAA** [comment]
@BrianCurtis-NOAA FYI, starting testing on this PR, fv3 sub pr has approvals, but still needs one from a required reviewer before merging after testing.

**Comment by BrianCurtis-NOAA** [comment]
> @BrianCurtis-NOAA FYI, starting testing on this PR, fv3 sub pr has approvals, but still needs one from a required reviewer before merging after testing.

OK, I can start on WCOSS2 once the ORT has passed. I've approved the FV3 PR so it has 2 reviews for merge. There are still conversations that need to be resolved there before merge.

**Comment by FernandoAndrade-NOAA** [comment]
Jenkins ci log attached, ORTs passed.
[ufs-weather-model » ort-docker-pipeline » PR-1823#1 Console [Jenkins].txt](https://github.com/ufs-community/ufs-weather-model/files/12545778/ufs-weather-model.ort-docker-pipeline.PR-1823.1.Console.Jenkins.txt)

**Comment by BrianCurtis-NOAA** [comment]
@jkbk2004 who is assigned to this PR?

**Comment by FernandoAndrade-NOAA** [comment]
It looks like Acorn will be skipped for this PR, @junwang-noaa please go ahead and resolve/mark resolved conversations in the sub PRs as well as in this PR.

**Comment by junwang-noaa** [comment]
@FernandoAndrade-NOAA Done

**Comment by DeniseWorthen** [comment]
@FernandoAndrade-NOAA Are we ready for subcomponent PR merges?

**Comment by FernandoAndrade-NOAA** [comment]
@DeniseWorthen yes everything looks good to go, please continue with the merge process

**Comment by DeniseWorthen** [comment]
CICE is ready, hash is [2ed3c05](https://github.com/NOAA-EMC/CICE/commit/2ed3c05c3c515eb70af3a726ff392283af97c4a5)

**Comment by jiandewang** [comment]
MOM6 merged, hash # is be40a4136

**Comment by jkbk2004** [comment]
FV3 merged as well, new hash https://github.com/NOAA-EMC/fv3atm/commit/a9fa26e3c03b2ae4d9294225c043ffd962028e72

**Comment by junwang-noaa** [comment]
I will update my UWM branch

**Comment by junwang-noaa** [comment]
Updated.

**Review by jkbk2004 — APPROVED** [review]

**Review by DeniseWorthen — APPROVED** [review]

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Pull request merged**
